src/main/java/seedu/address/model/deliveryman/Deliveryman.java

```diff
@@ -4,17 +4,32 @@
 
 import java.util.Objects;
 
+import seedu.address.model.IdObject;
 import seedu.address.model.person.Name;
 
 /**
  * Represents a Deliveryman in FoodZoom.
  * Guarantees: has a name that is unique.
  */
-public class Deliveryman {
+public class Deliveryman extends IdObject {
+    /** Tracker for assigning id. */
+    private static int idCounter = 0;
+
     private final Name name;
     //TODO:  add field for orders/routes attached to deliveryman.
 
     public Deliveryman(Name name) {
+        super(idCounter++);
```

Comment:
Consider using `ModelManager` to increment the `idCounter`. So you can just pass in only `id` and `name`.